django_freeradius/api/serializers.py

```diff
@@ -20,3 +22,36 @@ class Meta:
         fields = ['username', 'password', 'reply',
                   'called_station_id',
                   'calling_station_id']
+
+
+class RadiusAccountingSerializer(serializers.ModelSerializer):
+    framed_ip_address = serializers.IPAddressField(required=False, allow_blank=True)
+    session_time = serializers.IntegerField(required=False, default=0)
+    stop_time = serializers.DateTimeField(required=False)
+    update_time = serializers.DateTimeField(required=False)
+    # this is needed otherwise serialize will ignore acct_status_type from accounting packet
+    # as it's not a model field
+    acct_status_type = serializers.CharField()
+
+    def validate(self, data):
+        """
+        We need to set some timestamps according to the accounting packet type
+        * update_time: set everytime a Interim-Update / Stop packet is received
+        * stop_time: set everytime a Stop packet is received
+        * session_time: calculated if not present in the accounting packet
+        :param data: accounting packet
+        :return: Dict accounting packet
+        """
+        if data['acct_status_type'] == 'Interim-Update':
+            data['update_time'] = timezone.now()
+        if data['acct_status_type'] == 'Stop':
+            data['update_time'] = timezone.now()
+            data['stop_time'] = timezone.now()
+        if data['session_time'] is None:
+            data['session_time'] = (timezone.now() - data['start_time']).seconds
```

Comment:
instead of repeating `timezone.now()` you can call it once at the beginning store its value in an auxiliary variable that you then reuse